base_rest/restapi.py

```diff
@@ -333,3 +335,71 @@ def to_json_schema(self, service, direction):
         if self._unique_items is not None:
             json_schema["uniqueItems"] = self._unique_items
         return json_schema
+
+
+class MultipartFormData(RestMethodParam):
+    def __init__(self, parts):
+        """This allows to create multipart/form-data endpoints.
+        :param parts:  list of RestMethodParam
+        """
+        if not isinstance(parts, dict):
+            raise ValidationError(_("You must provide a dict of RestMethodParam"))
+        self._parts = parts
+
+    def to_openapi_properties(self, service, direction):
+        properties = {}
+        for key, part in self._parts.items():
+            if isinstance(part, BinaryData):
+                properties[key] = part.to_json_schema()
+            else:
+                properties[key] = part.to_json_schema(service, direction)
```

Comment:
@PierrickBrun If we rely on a spécific method here with `to_json_schema`, The signature must be the same for all the params and must be declared into `RestMethodParam`. Moreover, we should enforce the method signature defined into `RestMethodParam' by taking advantage of the `abstractmethod` decorator available on abc.ABC class.

```python

import abc

...

class RestMethodParam(abc.ABC):
     @abc.abstractmehod
    def from_params(self, service, params) -> Any:
        """
        This method is called to process the parameters received at the
        controller. This method should validate and sanitize these paramaters.
        It could also be used to transform these parameters into the format
        expected by the called method
        :param service:
        :param request: `HttpRequest.params`
        :return: Value into the format expected by the method
        """

     @abc.abstractmehod
    def to_response(self, service, result) -> http.Response:
        """
        This method is called to prepare the result of the call to the method
        in a format suitable by the controller (http.Response or JSON dict).
        It's responsible for validating and sanitizing the result.
        :param service:
        :param obj:
        :return: http.Response or JSON dict
        """
     @abc.abstractmehod
    def to_openapi_query_parameters(self, service) > dict:
        ...

     @abc.abstractmehod
    def to_openapi_requestbody(self, service) -> dict:
        ...

     @abc.abstractmehod
    def to_openapi_responses(self, service) -> dict:
        ...

     @abc.abstractmehod
    def to_json_schema(service, direction) -> dict:
        ...
```

Reply:
@lmignon done